presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java

```diff
@@ -147,6 +162,34 @@ public void close()
                 }
             }
         }
+        executor.shutdownNow();
+    }
+
+    @Override
+    public synchronized boolean resubmit(Verification verification, QueryException queryException)
+    {
+        if (!shouldRequeue(queryException)) {
+            return false;
+        }
+
+        VerificationRef ref = new VerificationRef(verification);
+        Integer requeuedCount = requeuedCounts.getOrDefault(ref, 0);
+        if (requeuedCount >= verificationRequeueLimit) {
+            log.info("Verification failed with %s, re-queue limit exceeded", queryException.getErrorCode());
+            return false;
+        }
+
+        queriesSubmitted.addAndGet(1);
+        requeuedCounts.compute(ref, (key, count) -> count == null ? 1 : count + 1);
+        completionService.submit(verification::run);
+        log.info("Verification failed with %s, re-queued for verification (%s/%s)", queryException.getErrorCode(), requeuedCount, verificationRequeueLimit);
```

Comment:
Without some kind of identifier I don't find this information useful. It just says that some query failed with some error code and we retried it.

Reply:
Added the name of the source query